File: Queues/MovingAverage/MovingAverage.py

class MovingAverage(object):

    def __init__(self, size):
        """
        :type size: int
        """
        self.size = size
        self.queue = []
        

    def next(self, val):
        """
        :type val: int
        :rtype: float
        """
        if len(self.queue) == self.size:
            self.queue.pop(0)
        self.queue.append(val)
        
        return sum(self.queue)/float(len(self.queue))
        
        
        

File: Queues/MovingAverage/MovingAverage.scala
import scala.collection.mutable.ArrayBuffer

class MovingAverage(_size: Int) {
    
    var window = new ArrayBuffer[Int]
    var sum = 0D
    
    def next(`val`: Int): Double = {
        if (window.size == _size) {

          sum -= window.remove(0)
        }
        
        window += `val`
        sum += `val`

        sum / window.length
    }

}